protos/feast/core/CoreService.proto

```diff
@@ -22,139 +22,49 @@ option go_package = "github.com/gojek/feast/protos/generated/go/feast/core";
-    option deprecated = true;
-  }
-  // Get all storage specs.
-  // This process returns a list of storage specs.
-  rpc ListStorage(google.protobuf.Empty) returns (CoreServiceTypes.ListStorageResponse) {
-    option deprecated = true;
-  }
-  // Get features specified in request.
-  // This process returns a list of feature specs.
-  rpc GetFeatures(CoreServiceTypes.GetFeaturesRequest) returns (CoreServiceTypes.GetFeaturesResponse);
-  // Get all features.
-  // This process returns a list of entity specs.
-  rpc ListFeatures(google.protobuf.Empty) returns (CoreServiceTypes.ListFeaturesResponse);
-  // Register a new feature to the metadata store, or update an existing feature.
-  // If any validation errors occur, only the first encountered error will be returned.
-  rpc ApplyFeature(feast.specs.FeatureSpec) returns (CoreServiceTypes.ApplyFeatureResponse);
-  // ApplyFeatures creates or updates a list of features in Feast.
-  // Updates only work for adding feature(s) to an existing entity.
-  // Changing data type of an existing feature is not supported and will result in an error.
-  rpc ApplyFeatures(CoreServiceTypes.ApplyFeaturesRequest) returns (CoreServiceTypes.ApplyFeaturesResponse);
-  // Register a new feature group to the metadata store, or update an existing feature group.
-  // If any validation errors occur, only the first encountered error will be returned.
-  rpc ApplyFeatureGroup(feast.specs.FeatureGroupSpec) returns (CoreServiceTypes.ApplyFeatureGroupResponse);
-  // Register a new entity to the metadata store, or update an existing entity.
-  // If any validation errors occur, only the first encountered error will be returned.
-  rpc ApplyEntity(feast.specs.EntitySpec) returns (CoreServiceTypes.ApplyEntityResponse);
-  // Retrieve the topic and topic configuration (such as the URI of the message brokers)
-  // for the client to write the feature values of an entity.
-  rpc GetTopic(CoreServiceTypes.GetTopicRequest) returns (CoreServiceTypes.GetTopicResponse);
-  // Request a signed URL where a Feast client can upload a feature values file for an import job.
-  // The signed URL will be valid by default for 5 minutes during which the client can start
-  // uploading the feature values file. As of 2019-06-28, only CSV and JSON files are supported,
-  // and the upload must complete in one PUT request i.e. resumable upload is not supported.
-  rpc GetUploadUrl(CoreServiceTypes.GetUploadUrlRequest) returns (CoreServiceTypes.GetUploadUrlResponse);
+    // Retrieve feature set details for a specific feature set
+    rpc GetFeatureSet (CoreServiceTypes.GetFeatureSetRequest) returns (CoreServiceTypes.GetFeatureSetResponse);
+
+    // Idempotent creation of feature set. Will not create a new feature set if schema does not change
+    rpc ApplyFeatureSet (CoreServiceTypes.ApplyFeatureSetRequest) returns (CoreServiceTypes.ApplyFeatureSetResponse);
 }
 
 message CoreServiceTypes {
```

Comment:
Do you think we can remove the nesting here? I feel the wrapper `CoreServiceTypes` is a bit unnecessary.

Reply:
I am fine with either. @zhilingc thoughts?

Reply:
Sure